src/Magnum/Audio/Listener.h

```diff
@@ -0,0 +1,247 @@
+
+/**
+@brief Base of @ref Listener2D and @ref Listener3D
+
+Contains methods common for all Listeners.
+
+## Usage
+
+Minimal scene setup with a listener will require the following code:
+
+@code
+Scene3D scene;
+Object3D object{&scene};
+Listener3D listener{object};
+
+// Important!
+listener.setActive();
+
+// ... and every frame, update listener to object transformation if necessary
+listener.update();
+
+@endcode
+
+Without calling `listener.setActive()`, calling `listener.update()` will not update the OpenAL listener.
+For two dimensional scenes simply replace `3D` with `2D`.
+
+## Active Listener
+
+There can only be at the most *one* active listener at a given time. This is because OpenAL only supports notion of one listener.
+Having multiple @ref Listener2D or @ref Listener3D instances can still be useful for conveniently switching between them for
+cinematics for example.
+
+## Sound Transformation
+
+@ref AbstractListener::setSoundTransformation() enables you to set transformation which is applied to the listeners orientation and position
+before passed onto OpenAL. This can be used for rotating two dimensional scenes as planes into three dimensional space or even scaling the audio
+scene to match a certain world scale. In the later case you might want to instead consider @ref Renderer::setSpeedOfSound().
+
+*/
+template <UnsignedInt dimensions, class Derived> class AbstractListener: public SceneGraph::AbstractFeature<dimensions, Float>, public AbstractListenerBase {
+        friend class ActiveListenerHolder;
+
+    public:
+        /** @brief Constructor */
+        explicit AbstractListener(SceneGraph::AbstractObject<dimensions, Float>& object):
+            SceneGraph::AbstractFeature<dimensions, Float>{object}
+        {
+            SceneGraph::AbstractFeature<dimensions, Float>::setCachedTransformations(SceneGraph::CachedTransformation::Absolute);
+        }
+
+        /** @brief Destructor */
+        virtual ~AbstractListener() {}
+
+        /** @brief Sound transformation */
+        const Matrix3x3& soundTransformation() const {
+            return _soundTransformation;
+        }
+
+        /**
+         * @brief Set sound transformation
+         * @param soundTransformation Transformation for transforming from world to listener space
+         * @return Reference to self (for method chaining)
+         */
+        Derived& setSoundTransformation(const Matrix3x3& soundTransformation) {
+            _soundTransformation = soundTransformation;
+            SceneGraph::AbstractFeature<dimensions, Float>::object().setDirty();
+            return *this;
+        }
+
+        /**
+         * @brief Update the listener
+         *
+         * Calls @ref SceneGraph::AbstractObject::setClean() on its parent object.
+         */
+        void update() { SceneGraph::AbstractFeature<dimensions, Float>::object().setClean(); } // TODO: Only if active?
+
+    private:
+        Matrix3x3 _soundTransformation;
+        Float _gain;
```

Comment:
Oh, right, so it would be for `Audio::Renderer::setListenerGain()`? Useful thing to have :)